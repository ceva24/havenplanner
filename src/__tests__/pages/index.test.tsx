import { render } from "@testing-library/react";
import { InferGetStaticPropsType } from "next";
import Index, { getStaticProps } from "../../pages/index";

const initialCharacter: Character = {
    name: "",
    experience: 0,
    gold: 0,
    notes: "",
};

const characterClasses: CharacterClass[] = [
    {
        id: 0,
        name: "Test 1",
        characterMatImageUrl:
            "/worldhaven/images/character-mats/gloomhaven/gh-brute.png",
    },
];

jest.mock("../../utils/data-loader", () => {
    return {
        loadCharacterClasses: jest.fn().mockImplementation(() => {
            return characterClasses;
        }),
    };
});

describe("Index", () => {
    it("renders", () => {
        const { asFragment } = render(
            <Index
                initialCharacter={initialCharacter}
                characterClasses={characterClasses}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});

describe("getStaticProps", () => {
    it("loads character class data", async () => {
        const data: InferGetStaticPropsType<typeof getStaticProps> =
            await getStaticProps({});

        expect(data.props.characterClasses).toEqual(characterClasses);
    });
});
