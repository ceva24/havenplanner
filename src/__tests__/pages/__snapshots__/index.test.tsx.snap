// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders 1`] = `
<DocumentFragment>
  <header
    class="MuiBox-root css-0"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter css-11s2mp2-MuiTypography-root"
    >
      Gloomhaven Character Planner
    </h1>
  </header>
  <main
    class="MuiContainer-root css-ur2jdm-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-10wqj5h"
    >
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        style="min-width: 10rem;"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-b805e-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          id="select-class-label"
        >
          Class
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-16rb7a0-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="select-class-label"
            class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-v3zyv7-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1wu7ecg-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-9425fu-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                Class
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10 css-1in6hls-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      />
    </div>
  </main>
  <header
    class="MuiBox-root css-0"
  >
    <h1
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-hfkqo3-MuiTypography-root"
    >
      <em>
        Gloomhaven and all related properties, images and text are owned by Cephalofair Games.
      </em>
    </h1>
  </header>
</DocumentFragment>
`;
